cannon/mipsevm/testutil/elf.go

```diff
@@ -10,15 +10,18 @@ import (
 	"github.com/ethereum-optimism/optimism/cannon/mipsevm/program"
 )
 
-func LoadELFProgram[T mipsevm.FPVMState](t *testing.T, name string, initState program.CreateFPVMState[T]) T {
+func LoadELFProgram[T mipsevm.FPVMState](t *testing.T, name string, initState program.CreateInitialFPVMState[T], doPatchGo bool) T {
 	elfProgram, err := elf.Open(name)
 	require.NoError(t, err, "open ELF file")
 
 	state, err := program.LoadELF(elfProgram, initState)
 	require.NoError(t, err, "load ELF into state")
 
-	err = program.PatchGo(elfProgram, state)
-	require.NoError(t, err, "apply Go runtime patches")
+	if doPatchGo {
```

Comment:
Currently skipping any go-specific patching for MTCannon, but we can add this back in after testing the performance, etc.

Reply:
This is OK. Just one note: `PatchGo` disables the gc too. Might also be worth breaking it up into GC patches and performance/misc related ones.